### PR TITLE
Add Sha224 digest, HMAC-SHA224, and SHA224 KAT

### PR DESCRIPTION
**Pull request by bencivjan**
### Description of changes: 
- Added SHA224 to supported digests
- HMAC with SHA224 now offered

### Testing:
- NIST verified SHA224 KAT

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: aws-lc-rs/src/digest/sha.rs [hunk 1]
```diff
@@ -84,6 +87,23 @@ pub static SHA1_FOR_LEGACY_USE_ONLY: Algorithm = Algorithm {
...
 
+/// SHA-256 as specified in [FIPS 180-4].
+///
+/// [FIPS 180-4]: http://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf
+#[allow(deprecated)]
+pub static SHA224: Algorithm = Algorithm {
+    output_len: SHA224_OUTPUT_LEN,
+    chaining_len: SHA224_OUTPUT_LEN,
+    block_len: 512 / 8,
+    max_input_len: SHA256_MAX_INPUT_LEN,
```
Comment (justsmth) [inline aws-lc-rs/src/digest/sha.rs:98]: We should have a `SHA224_MAX_INPUT_LEN`.  I also noticed that the SHA384 calues below need to be updated as well.

**Review by justsmth — APPROVED** [review]

File: aws-lc-rs/src/digest/sha.rs [hunk 2]
```diff
@@ -109,7 +138,7 @@ pub static SHA384: Algorithm = Algorithm {
     output_len: SHA384_OUTPUT_LEN,
     chaining_len: SHA512_OUTPUT_LEN,
     block_len: SHA512_BLOCK_LEN,
```
Comment (justsmth) [inline aws-lc-rs/src/digest/sha.rs:140]: Nit: SHA384 has the same block/chaining length as SHA512, so the values configured are correct.  A comment here indicating that would make it clear that this is not a typo.

File: aws-lc-rs/src/digest/sha.rs [hunk 3]
```diff
@@ -84,6 +96,23 @@ pub static SHA1_FOR_LEGACY_USE_ONLY: Algorithm = Algorithm {
...
 };
 
+/// SHA-256 as specified in [FIPS 180-4].
+///
+/// [FIPS 180-4]: http://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf
+#[allow(deprecated)]
+pub static SHA224: Algorithm = Algorithm {
+    output_len: SHA224_OUTPUT_LEN,
+    chaining_len: SHA224_OUTPUT_LEN,
+    block_len: 512 / 8,
```
Comment (justsmth) [inline aws-lc-rs/src/digest/sha.rs:106]: Nit: It's not clear why we don't have a constant defined for this value like we do for the other parameters. (The same in SHA256.)

File: aws-lc-rs/src/digest/sha.rs [hunk 4]
```diff
@@ -84,6 +96,23 @@ pub static SHA1_FOR_LEGACY_USE_ONLY: Algorithm = Algorithm {
     hash_nid: NID_sha1,
 };
 
+/// SHA-256 as specified in [FIPS 180-4].
+///
+/// [FIPS 180-4]: http://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf
+#[allow(deprecated)]
+pub static SHA224: Algorithm = Algorithm {
+    output_len: SHA224_OUTPUT_LEN,
+    chaining_len: SHA224_OUTPUT_LEN,
```
[the remark this answers is not in the document]
Comment (bencivjan) [reply aws-lc-rs/src/digest/sha.rs:105]: Yeah you're right

**Review by justsmth — APPROVED** [review]
The "api-diff" CI appears to be triggered by the change to `SHA1_OUTPUT_LEN`, but I believe this is a false positive.

```
Changed items in the public API
===============================
-pub const aws_lc_rs::digest::SHA1_OUTPUT_LEN: OUTPUT_LEN
+pub const aws_lc_rs::digest::SHA1_OUTPUT_LEN: _
```

File: aws-lc-rs/src/digest/sha.rs [hunk 5]
```diff
@@ -31,12 +36,21 @@ pub const SHA512_256_OUTPUT_LEN: usize = 256 / 8;
...
 #[allow(clippy::cast_possible_truncation)]
 const SHA256_MAX_INPUT_LEN: u64 = u64::MAX;
 
 /// SHA-384, SHA-512, and SHA-512/256 are limited to an input size of 2^128-1 bits according to the spec.
 /// u64 is more than sufficient enough for practical usecases, so we limit the input length to 2^64-1 bits.
+#[allow(clippy::cast_possible_truncation)]
+const SHA384_MAX_INPUT_LEN: u64 = u64::MAX;
+
 #[allow(clippy::cast_possible_truncation)]
 const SHA512_MAX_INPUT_LEN: u64 = u64::MAX;
```
Comment (samuel40791765) [inline aws-lc-rs/src/digest/sha.rs:55]: Could we just combine these into a single variable called `SHA2_MAX_INPUT_LEN` or `DIGEST_MAX_INPUT_LEN` and keep the comments above it?

Comment (bencivjan) [reply aws-lc-rs/src/digest/sha.rs:55]: Yeah

**Review by samuel40791765 — APPROVED** [review]

**Review by justsmth — APPROVED** [review]

**Pull request merged**